python_ta/debug/snapshot.py

```diff
@@ -47,7 +58,27 @@ def snapshot():
 
         frame = frame.f_back
 
-    return variables
+    if save:
+        json_compatible_vars = snapshot_to_json(variables)
+        command = ["npx", "memory-viz@" + memory_viz_version]
```

Comment:
As a pre-processing step, if `memory_viz_version` is not `"latest"`, parse the version information (https://packaging.pypa.io/en/latest/version.html#packaging.version.Version). If less than 0.3.1, print a warning message (use the `logging` module, see other parts of PythonTA) saying that PythonTA only supports MemoryViz versions 0.3.1 and later.